io-engine/src/lvm/vg_pool.rs

```diff
@@ -0,0 +1,294 @@
+        vgs.into_iter().next().ok_or(Error::NotFound {
+            query: QueryArgs(args).query(),
+        })
+    }
+
+    /// List all the volume groups using the provided list options.
+    pub(crate) async fn list(
+        opts: &CmnQueryArgs,
+    ) -> Result<Vec<VolumeGroup>, Error> {
+        let mut args = vec![
+            "--units=b",
+            "--nosuffix",
+            "-q",
+            "--options=vg_name,vg_uuid,vg_size,vg_free,vg_tags,pv_name",
+            "--report-format=json",
+        ];
+        let select = QueryArgs::query_args(opts);
+        let select_query = format!("--select={select}");
+        if !select.is_empty() {
+            args.push(select_query.trim_end_matches(','));
+        }
+        let report: VolGroups =
+            LvmCmd::vg_list().args(args.as_slice()).report().await?;
+
+        let vgs = report
+            .vg
+            .into_iter()
+            // todo: not needed as we did the select?
+            .filter(|vg| vg.matches(opts))
+            .collect::<Vec<_>>();
+
+        Ok(vgs)
+    }
+
+    /// Import a volume group with the name provided or create one with the name
+    /// and disks provided currently only import is supported.
+    pub(crate) async fn import_or_create(
+        args: PoolArgs,
+    ) -> Result<VolumeGroup, Error> {
+        let pool = Self::import(args).await?;
+        info!("The lvm vg pool '{}' has been created", pool.name());
+        Ok(pool)
+    }
+
+    /// Import a volume group by its name, match the disks on the volume group
+    /// and if true add our tag to the volume group to make it available
+    /// as a Pool.
+    pub(crate) async fn import(args: PoolArgs) -> Result<VolumeGroup, Error> {
+        let name = &args.name;
+        let mut vg = Self::lookup(CmnQueryArgs::any().named(name)).await?;
+
+        if args.uuid.is_some() {
+            return Err(Error::VgUuidSet {});
+        }
+
+        if vg.disks != args.disks {
+            return Err(Error::DisksMismatch {
+                args: args.disks,
+                vg: vg.disks,
+            });
+        }
+
+        if vg.tags.contains(&Property::Lvm) {
+            return Ok(vg);
+        }
+
+        LvmCmd::vg_change(name)
+            .arg("-q")
+            .tag(Property::Lvm)
+            .run()
+            .await?;
+        vg.tags.push(Property::Lvm);
+        Ok(vg)
+    }
+
+    /// Delete the volume group.
+    pub(crate) async fn destroy(mut self) -> Result<(), Error> {
+        // As currently only import of volume group is supported
+        // exporting the volume group on destroy.
+        let name = self.name().to_string();
+        self.export().await?;
+        info!("LVM pool '{name}' has been destroyed successfully");
+        self.ptpl().destroy().ok();
+        Ok(())
+    }
+
+    /// Exports the volume by removing our tag from it.
+    /// todo: export the LV's as well...
+    pub(crate) async fn export(&mut self) -> Result<(), Error> {
```

Comment:
Is Property::Lvm the only property that needs to be cleared from VG, and other properties are LV specific?

Reply:
Export usually means just "shutdown" the pool so it can be imported elsewhere. Probably we shouldn't even remove our tag from it?

Reply:
Since we doing export upon `destroy` as well here, the same disks can probably be used for an lvs pool as well after this, right? In that case it makes sense to cleanup whatever could be left behind from lvm pool, even though lvs pool will write it's own metadata.

Reply:
> That should be as part of destroy and not export though.

Yeah, destroy = cleanup + export

Yes the most obvious logical next step should now be to integrate create and destroy. Until then, this should remain experimental(with a disclaimer). Do we plan to add some documentation/howto at this point w.r.t this feature?

Reply:
We now do create and destroy